### ISSUE TITLE
bug: browser autofill background color overlaps label in md mode

### ISSUE THREAD

**Issue by abennouna** [ionic-team/ionic-framework#18141]
# Bug Report

**Ionic version:**
[x] **4.x**

**Current behavior:**
Starting from 4.3.0, **autofilled** inputs overlap their floating labels in MD mode.

**Expected behavior:**
As for non-autofilled inputs, floating labels must stay entirely visible.

**Steps to reproduce:**
An `ion-item` containing a **floating** `ion-label` and an `ion-input` that is auto-fillable and auto-filled.

**Related code:**
N/A

**Other information:**
***Ionic Screenshots***
_MD_
![TXbNfO3](https://user-images.githubusercontent.com/1470318/56816672-eff9de80-683b-11e9-8de4-99f7c4023ae2.png)

_iOS_
![vqAPHmu](https://user-images.githubusercontent.com/1470318/56816676-f38d6580-683b-11e9-8c3e-6b5fb4f1cf34.png)

***MD implementation Screenshots***
_Chrome_
![LqzQAHJ](https://user-images.githubusercontent.com/1470318/56816689-f8eab000-683b-11e9-8c13-55904ab85b74.png)

_Safari_
![xpuBg2c](https://user-images.githubusercontent.com/1470318/56816700-fdaf6400-683b-11e9-80e5-8258c1bfd3ce.png)

**Ionic info:**

```
Ionic:

   ionic (Ionic CLI)             : 4.12.0 (~/.nvm/versions/node/v10.12.0/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.3.0
   @angular-devkit/build-angular : 0.13.8
   @angular-devkit/schematics    : 7.3.8
   @angular/cli                  : 7.3.8
   @ionic/angular-toolkit        : 1.5.1

System:

   NodeJS : v10.12.0 (~/.nvm/versions/node/v10.12.0/bin/node)
   npm    : 6.4.1
   OS     : macOS Mojave
```

**Comment by EricKit** [comment]
I have this issue with an ion-label and ion-input with md and iOS with the label set to stacked.

**Comment by mebibou** [comment]
Yes this issue also appears on `position="stacked"`. It seems the style
```
.native-input.sc-ion-input-md:-webkit-autofill {
    background-color: transparent;
}
```
gets overridden. Could you guys simply put a `!important` at the end to fix this ugly issue? how can we put this in prod?

![Screen Shot 2019-09-16 at 7 06 07 PM](https://user-images.githubusercontent.com/305342/64978128-1a186f80-d8b5-11e9-84d9-f6067e7ebe37.png)

**Comment by jaipresto** [comment]
Until Ionic sort this (if they can; it's deeply-rooted and seems to be browser-specific), a hack is to use a combination of `height` and `box-shadow`. Height prevents it covering the label, and a box-shadow to change the color.

So an example using code from @mebibou:
```
.native-input.sc-ion-input-md:-webkit-autofill {
    height: 0px;
    box-shadow: 0 0 0 10px #ebe6ff inset;
}
```

or something more general, as per [these answers](https://stackoverflow.com/questions/2338102/override-browser-form-filling-and-input-highlighting-with-html-css):
```
input:-webkit-autofill {
    height: 0px;
    -webkit-box-shadow:0 0 0 10px white inset;
    -webkit-text-fill-color: black;
    font-weight: 500px;
}

input:-webkit-autofill:focus {
    height: 0px;
    -webkit-box-shadow: 0 0 0 10px white inset;
    -webkit-text-fill-color: black;
    font-weight: 500px;
} 
```

Just change the color of the box-shadow property to anything you want, and **ensure you add the `inset` value**. You can also do other stuff like change the font-family, add a transform, and even do some animation.

Of course the box-shadow doesn't get rid of the default browser css, it just sits on top.

**Comment by bitflower** [comment]
I have had this for a long time but finally found the time to contribute just to find this issue ;-)

<img width="418" alt="Bildschirmfoto 2020-03-29 um 19 25 33" src="https://user-images.githubusercontent.com/6666636/77988646-28901b00-731d-11ea-8461-4d95d516b6d8.png">

**Comment by dylanvdmerwe** [comment]
Note that this is still a thing. @ionic/angular: 5.6.9

Autofill when used on a desktop browser looks shocking. 

![image](https://user-images.githubusercontent.com/2852839/121418753-8e3d5580-c96b-11eb-8bd5-da0aafb89dc3.png)

**Comment by DajuSar** [comment]
its 2021, 2 years from the last comment and this error still happens.
![image](https://user-images.githubusercontent.com/55637730/145490945-3ae00647-bd5f-4c14-b5cc-480ceaeffe73.png)

**Comment by sean-perkins** [comment]
@DajuSar judging open issues based on their reported date can be a difficult metric. While we'd always hope to resolve issues quickly; there's always a list of changing priorities that the team must manage. 

Issues that are less of a functional issue, are great issues for community contributions. 

Seeing as this issue is older, I'd love assistance from the community on debugging and shaping the resolution. I've opened #24443 to introduce a fix to the documented issues in this thread. The solution will apply a consistent appearance to the item, when a form control has autofill; regardless of iOS/md/etc.

Here's a quick preview of that appearance:

<img src="https://user-images.githubusercontent.com/13732623/146664538-7143564e-c881-4416-84a0-d531dde98a56.png" width="350px" />

I'd appreciate feedback on if this hits the mark for the struggles you are facing. Thanks!

**Comment by liamdebeasi** [comment]
Hi everyone,

We recently introduced a new form component architecture in Ionic 7. As part of this, we updated the internals such that the input and the floating label should no longer overlap. We have more information available in https://ionic.io/blog/announcing-the-ionic-7-beta.

Could someone give the Ionic 7 beta a try and let me know if the resolution works for their use case? To get the new architecture you will need to use the `label` property on `ion-input` or `ion-textarea`:

```html
<ion-input label="Email:"></ion-input>
```
